examples/newsreaderExample/src/main/java/io/realm/examples/newsreader/model/network/NYTimesDataLoader.java

```diff
@@ -0,0 +1,119 @@
+import io.realm.examples.newsreader.model.DefaultTransactionCallback;
+import io.realm.examples.newsreader.model.entity.NYTimesStory;
+import retrofit.JacksonConverterFactory;
+import retrofit.Retrofit;
+import retrofit.RxJavaCallAdapterFactory;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.schedulers.Schedulers;
+import rx.subjects.BehaviorSubject;
+import timber.log.Timber;
+
+/**
+ * Class that handles network requests for the New York Times API
+ */
+public class NYTimesDataLoader {
+
+    private NYTimesService nyTimesService;
+    private SimpleDateFormat inputDateFormat = new SimpleDateFormat("yyyy-MM-d'T'HH:mm:ssZZZZZ", Locale.US);
+    private SimpleDateFormat outputDateFormat = new SimpleDateFormat("MM-dd-yyyy", Locale.US);
+    private String apiKey;
+    private Realm realm;
+    private BehaviorSubject<Boolean> networkInUse;
+
+    public NYTimesDataLoader() {
+        Retrofit retrofit = new Retrofit.Builder()
+                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
+                .addConverterFactory(JacksonConverterFactory.create())
+                .baseUrl("http://api.nytimes.com/")
+                .build();
+        nyTimesService = retrofit.create(NYTimesService.class);
+    }
+
+    public void loadData(String sectionKey, String apiKey, Realm realm, BehaviorSubject<Boolean> networkLoading) {
+        this.apiKey = apiKey;
+        this.realm = realm;
+        this.networkInUse = networkLoading;
+        loadNextSection(sectionKey);
+    }
+
+    // Load all sections one by one.
+    private void loadNextSection(@NonNull final String sectionKey) {
+        networkInUse.onNext(true);
+        nyTimesService.topStories(sectionKey, apiKey)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new Action1<NYTimesResponse<List<NYTimesStory>>>() {
+                    @Override
+                    public void call(NYTimesResponse<List<NYTimesStory>> response) {
+                        Timber.d("Success - Data received: %s", sectionKey);
+                        processAndAddData(realm, response.section, response.results);
+                        networkInUse.onNext(false);
+                    }
+                }, new Action1<Throwable>() {
+                    @Override
+                    public void call(Throwable throwable) {
+                        networkInUse.onNext(false);
+                        Timber.d("Failure: Data not loaded: %s - %s", sectionKey, throwable.toString());
+                    }
+                });
+    }
+
+    // Converts data into a usable format and save it to Realm
+    private void processAndAddData(final Realm realm, final String sectionKey, final List<NYTimesStory> stories) {
+        if (stories.isEmpty()) return;
+
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
```

Comment:
I am thinking maybe it worth to add a interface `Transaction.execute(RealmConfiguration config)` for async, cases like this don't have to maintain a Realm instance anymore.